src/controllers/handlers/community-voice-chat/community-voice-chat-handler.ts

```diff
@@ -0,0 +1,145 @@
+  }
+
+  if (!body.action || !Object.values(CommunityVoiceChatAction).includes(body.action)) {
+    throw new InvalidRequestError(
+      'The property action is required and must be one of: create, join, request-to-speak, promote-speaker, demote-speaker, kick-player'
+    )
+  }
+
+  const lowerCaseUserAddress = body.user_address.toLowerCase()
+
+  try {
+    switch (body.action) {
+      case CommunityVoiceChatAction.CREATE: {
+        logger.debug('Creating community voice chat credentials for moderator')
+        const credentials = await voice.getCommunityVoiceChatCredentialsForModerator(
+          body.community_id,
+          lowerCaseUserAddress
+        )
+        logger.debug(
+          `Created community voice chat credentials for moderator ${lowerCaseUserAddress} in community ${body.community_id}`
+        )
+        return {
+          status: 200,
+          body: {
+            connection_url: credentials.connectionUrl
+          }
+        }
+      }
+
+      case CommunityVoiceChatAction.JOIN: {
+        logger.debug('Joining community voice chat as member')
+        const credentials = await voice.getCommunityVoiceChatCredentialsForMember(
+          body.community_id,
+          lowerCaseUserAddress
+        )
+        logger.info(
+          `Community voice chat access granted for member ${lowerCaseUserAddress} in community ${body.community_id}`
+        )
+        return {
+          status: 200,
+          body: {
+            connection_url: credentials.connectionUrl
+          }
+        }
+      }
+
+      case CommunityVoiceChatAction.REQUEST_TO_SPEAK: {
+        logger.info(`User ${lowerCaseUserAddress} requesting to speak in community ${body.community_id}`)
+
+        await voice.requestToSpeakInCommunity(body.community_id, lowerCaseUserAddress)
+
+        return {
+          status: 200,
+          body: {
+            message: 'Request to speak sent successfully'
+          }
+        }
+      }
+
+      case CommunityVoiceChatAction.PROMOTE_SPEAKER: {
+        logger.info(`Promoting user ${lowerCaseUserAddress} to speaker in community ${body.community_id}`)
+
+        await voice.promoteSpeakerInCommunity(body.community_id, lowerCaseUserAddress)
+
+        return {
+          status: 200,
+          body: {
+            message: 'User promoted to speaker successfully'
+          }
+        }
+      }
+
+      case CommunityVoiceChatAction.DEMOTE_SPEAKER: {
+        logger.info(`Demoting user ${lowerCaseUserAddress} to listener in community ${body.community_id}`)
+
+        await voice.demoteSpeakerInCommunity(body.community_id, lowerCaseUserAddress)
+
+        return {
+          status: 200,
+          body: {
+            message: 'User demoted to listener successfully'
+          }
+        }
+      }
+
+      case CommunityVoiceChatAction.KICK_PLAYER: {
+        logger.info(`Kicking user ${lowerCaseUserAddress} from community ${body.community_id}`)
```

Comment:
This endpoint is correct, but if you want, there's an alternative to have an specific endpoint to remove users from the voice chat:
`DELETE /community-voice-chat/:communityId/users/:address` can be used to kick a player.